### PR TITLE
SQL language generator

### PR DESCRIPTION
**Pull request by emiliojazz**
Hello Neil, hello contributors.
Thanks for this fantastic piece of code, I used it successfully in various projects of mine.
I implemented an SQL language generator.
It's functional (tested as MySQL trigger) but it lacks the support for some looping functions.
Let me know what you think about it,
regards.

### CODE REVIEW

File: generators/sql.js [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        // http://php.net/manual/en/reserved.constants.php
+    'PHP_VERSION,PHP_MAJOR_VERSION,PHP_MINOR_VERSION,PHP_RELEASE_VERSION,PHP_VERSION_ID,PHP_EXTRA_VERSION,PHP_ZTS,PHP_DEBUG,PHP_MAXPATHLEN,PHP_OS,PHP_SAPI,PHP_EOL,PHP_INT_MAX,PHP_INT_SIZE,DEFAULT_INCLUDE_PATH,PEAR_INSTALL_DIR,PEAR_EXTENSION_DIR,PHP_EXTENSION_DIR,PHP_PREFIX,PHP_BINDIR,PHP_BINARY,PHP_MANDIR,PHP_LIBDIR,PHP_DATADIR,PHP_SYSCONFDIR,PHP_LOCALSTATEDIR,PHP_CONFIG_FILE_PATH,PHP_CONFIG_FILE_SCAN_DIR,PHP_SHLIB_SUFFIX,E_ERROR,E_WARNING,E_PARSE,E_NOTICE,E_CORE_ERROR,E_CORE_WARNING,E_COMPILE_ERROR,E_COMPILE_WARNING,E_USER_ERROR,E_USER_WARNING,E_USER_NOTICE,E_DEPRECATED,E_USER_DEPRECATED,E_ALL,E_STRICT,__COMPILER_HALT_OFFSET__,TRUE,FALSE,NULL,__CLASS__,__DIR__,__FILE__,__FUNCTION__,__LINE__,__METHOD__,__NAMESPACE__,__TRAIT__');
```
Comment (NeilFraser) [inline generators/sql.js]: These are not reserved words in SQL.

File: generators/sql.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+/**
+ * Order of operation ENUMs.
+ * http://php.net/manual/en/language.operators.precedence.php
```
Comment (NeilFraser) [inline generators/sql.js]: These order of operation ENUMs aren't SQL.

File: generators/sql.js [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+ * anything.  A trailing semicolon is needed to make this legal.
+ * @param {string} line Line of generated code.
+ * @return {string} Legal line of code.
+ */
+Blockly.SQL.scrubNakedValue = function(line) {
+  return line + ';\n';
+};
+
+/**
+ * Encode a string as a properly escaped PHP string, complete with
```
Comment (NeilFraser) [inline generators/sql.js]: PHP?

File: generators/sql.js [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+  return '\'' + string + '\'';
+};
+
+/**
+ * Common tasks for generating SQL from blocks.
+ * Handles comments for the specified block and any connected value blocks.
+ * Calls any statements following this block.
+ * @param {!Blockly.Block} block The current block.
+ * @param {string} code The SQL code created for this block.
+ * @return {string} PHP code with comments and subsequent blocks added.
```
Comment (NeilFraser) [inline generators/sql.js]: PHP?

File: generators/sql.js [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+ * @private
+ */
+Blockly.SQL.scrub_ = function(block, code) {
+  var commentCode = '';
+  // Only collect comments for blocks that aren't inline.
+  if (!block.outputConnection || !block.outputConnection.targetConnection) {
+    // Collect comment for this block.
+    var comment = block.getCommentText();
+    if (comment) {
+      commentCode += Blockly.SQL.prefixLines(comment, '// ') + '\n';
```
Comment (NeilFraser) [inline generators/sql.js]: I don't think SQL uses // comments.

File: generators/sql/colour.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+ * @author daarond@gmail.com (Daaron Dwyer)
+ */
+'use strict';
+
+goog.provide('Blockly.SQL.colour');
+
+goog.require('Blockly.SQL');
+
+
+Blockly.SQL['colour_picker'] = function(block) {
```
Comment (NeilFraser) [inline generators/sql/colour.js]: What does a colour_picker do in SQL?

**Comment by NeilFraser** [comment]
Correct me if I'm wrong, but I think you just did a search and replace for Blockly.PHP to Blockly.SQL, and changed the @author names.  I don't see any SQL code generation here, or SQL specific blocks.

**Comment by emiliojazz** [comment]
Yes, I started with Blockly.PHP class and then made edits on it.
I certainly forgot many places to substitute and adjust reserved keywords etc...but I can tell you that the code works. Of course I will correct the things...and maybe pull a new request if you are interested.
Thank you for your patience :-)

**Pull request closed without merging**